core/src/Dscp/Core/Foundation/Educator/Certificate/Meta.hs

```diff
@@ -0,0 +1,36 @@
+
+module Dscp.Core.Foundation.Educator.Certificate.Meta where
+
+import Data.Time.Calendar (Day (..))
+
+import Dscp.Core.Foundation.Educator.ItemDesc
+import Dscp.Core.Foundation.Educator.Certificate.EducationForm
+import Dscp.Util
+
+-- | Datatype containing info about a certificate issued by Educator.
+data CertificateMeta = CertificateMeta
+    { cmStudentName      :: !ItemDesc
+    , cmStudentBirthDate :: !Day
+    , cmStartYear        :: !Word16
+    , cmEndYear          :: !Word16
+    , cmEducationForm    :: !EducationForm
+    , cmNumber           :: !Natural
+    , cmIssueDate        :: !Day
+    , cmTitle            :: !ItemDesc
+    , cmMajor            :: !ItemDesc
+    , cmSpecialization   :: !(Maybe ItemDesc)
+    } deriving (Show, Eq, Generic)
+
+instance Buildable CertificateMeta where
+    build CertificateMeta {..} =
+        "{ studentName = "+|cmStudentName|+
+        ", studentBirthDate = "+|cmStudentBirthDate|+
+        ", startYear = "+|cmStartYear|+
+        ", endYear = "+|cmEndYear|+
+        ", educationForm = "+|cmEducationForm|+
+        ", number = "+|toInteger cmNumber|+
+        ", issueDate = "+|cmIssueDate|+
+        ", title = "+|cmTitle|+
+        ", major = "+|cmMajor|+
+        ", specialization = "+|cmSpecialization|+" }"
```

Comment:
Let me explain why personally I don't like such granularity of `Certificate` modules.
Dividing into these small pieces, AFAIU, you improve readability only a little bit, disregard whether a developer  navigates through a search in his IDE or opens files manually in a terminal - in any case accessing `Certificate.hs` module instead of `Certificate/*.hs` would be not more difficult. Covering all the module visually would also be yet easy as soon as all certificate-related type declarations fit into 1-2 screens.

On the other hand, efforts required for modification are increased significantly. If I want to add a type, I have to create a module for that and add several imports. I'm afraid to imagine the diff if one day we need to change the structure of certificate datatypes.

Even if number/size of certificate-related datatypes grow too large one day, IMO _that_ would the right time to make a split so that balance between readability and write efforts remains sane.

I would not carp if I was the only one who'd like to see those types in one module, but since @flyingleafe also thinks so, I can't ignore.